app/vmagent/remotewrite/remotewrite.go

```diff
@@ -411,6 +412,19 @@ func tryPush(at *auth.Token, wr *prompbmarshal.WriteRequest, forceDropSamplesOnF
 		return true
 	}
 
+	// Push metadata separately from time series, since it doesn't need to be sharding,
+	// and can skip relabeling, stream aggregation, deduplication, etc.
+	if !tryPushMetadataToRemoteStorages(rwctxs, mms, forceDropSamplesOnFailure) {
+		return false
+	}
```

Comment:
Do I understand it correctly that the following is correct?

- using multitenancy via headers will not be applied to metadata - it will be forwarded to tenant `0:0`
- dropping some series with relabeling will not cause metadata to be dropped

Reply:
>using multitenancy via labels will not be applied to metadata

Yes, it's a tricky one.
It's possible to try fetching tenant info from the metrics, but since different metrics can have different tenant labels in a single request, it would then require to match the metadata and metrics, which is expensive.
So currently, it won't be applied to metadata, I will add it to the doc.

>dropping some series with relabeling will not cause metadata to be dropped

I think we discussed whether relabeling or dropping metrics should affect the metadata. The consensus was to keep the initial version simple and improve it if necessary.

cc @hagen1778